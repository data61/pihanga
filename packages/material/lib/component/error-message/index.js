"use strict";

exports.__esModule = true;

var _errorMessage = require("./error-message.component");

Object.keys(_errorMessage).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  exports[key] = _errorMessage[key];
});