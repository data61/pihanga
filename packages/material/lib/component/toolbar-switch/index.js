"use strict";

exports.__esModule = true;

var _toolbarSwitch = require("./toolbar-switch.component");

Object.keys(_toolbarSwitch).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  exports[key] = _toolbarSwitch[key];
});

var _toolbarSwitch2 = require("./toolbar-switch.actions");

Object.keys(_toolbarSwitch2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  exports[key] = _toolbarSwitch2[key];
});