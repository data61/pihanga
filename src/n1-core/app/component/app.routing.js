export const APP_ROUTING = {
//  '/': '/login', // a redirect
  '/': '/cars', // a redirect
};
