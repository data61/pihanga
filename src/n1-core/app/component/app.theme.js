import { createMuiTheme } from '@material-ui/core/styles';

import purple from '@material-ui/core/colors/purple';
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';

export default (defTheme = createMuiTheme()) => {
  return createMuiTheme({
    palette: {
      primary: purple, // Purple and green play nicely together.
      secondary: {
        ...green,
        A400: '#00e677',
      },
      error: red,
    },  
    status: {
      danger: 'orange',
    },
    typography: {
      // fontFamily: '-apple-system,system-ui,BlinkMacSystemFont,' +
      //   '"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif',
      versionFootNote: {
        fontWeight: defTheme.typography.fontWeightMedium,
        fontSize: 10,
      }
    }
  });
};
