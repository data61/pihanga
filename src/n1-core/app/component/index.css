body {
  margin: 0;
  padding: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
}

#root {
  margin: 0;
  padding: 0;
}

a {
    color: #4078c0;
    text-decoration: none;
    cursor: pointer;
}

a:hover {
    text-decoration: underline;
}

/** dimming a div **/
.dimmed {
  position: relative;
}

.dimmed:after {
  content: " ";
  z-index: 10;
  display: block;
  position: absolute;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  background: rgba(255, 255, 255, 0.6);
}
