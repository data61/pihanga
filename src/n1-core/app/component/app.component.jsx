import React from 'react';
import { Provider, connect } from 'react-redux';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
//import { card } from '../../card.service';
import { Card } from 'n1-core';

import { reloadBackend } from './app.actions';
import theme from './app.theme';

const AppComponent = (store, routerComponentWrapper) => {
  const loadingRouteElement = (
    <div>
      LOADING
    </div>
  );

  const routeNotFoundElementFunc = invalidRoutePath => (
    <div>
      <p>ERROR: Path &quot;{invalidRoutePath}&quot; does not exist. Please check your URL.</p>
    </div>
  );

  const N1RouterComponent = connect(s => s)(
    routerComponentWrapper
      .customise(
        loadingRouteElement,
        routeNotFoundElementFunc,
      )
      .getRouterComponentConstructor(),
  );

  // PIHANAGA

  const cardNotFoundFunc = missingCard => (
    <div>
      <p>ERROR: Can't find card  &quot;{missingCard}&quot;. Please check your configuration.</p>
    </div>
  );

  // const PageComponent = connect(s => s)(s => {
  //   const page = s.pihanga.page;
  //   if (!page) {
  //     return cardNotFoundFunc("page");
  //   }
  //   const cardComponent = card(page.cardType);
  //   const state = Object.assign({}, s, {card: page, cardName: 'page'})
  //   const el = React.createElement(cardComponent, state);
  //   return el;
  // });

  reloadBackend();

  // return (
  //   <Provider store={store}>
  //     <MuiThemeProvider theme={theme()}>
  //       <N1RouterComponent />
  //     </MuiThemeProvider>
  //   </Provider>
  // );
  return (
    <Provider store={store}>
      <MuiThemeProvider theme={theme()}>
        <Card cardName={'page'} />
      </MuiThemeProvider>
    </Provider>
  );
};

export default AppComponent;
