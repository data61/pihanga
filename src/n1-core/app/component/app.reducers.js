import { update } from 'n1-core';
import { prettifyNumber } from 'n1-core/utils';

import {  ACTION_TYPES as APP_ACTION_TYPES } from './app.actions';

export default (registerReducer) => {
  registerReducer(APP_ACTION_TYPES.CREATE_SCRATCH, (state, action) =>  {
    const path = ['scratch', ...action.path];
    const s = update(state, path, action.initial || {});
    return s;
  });
}

export function reduceOnAcceptSignOut(state) {
  return update(state, [], {
    user: {},
    scratch: {},
  });
}

export function reduceOnSessionChecked(state) {
  return update(state, [], {
    checkedSession: true,
  });
}

export function reduceOnTimeOutSessionSoon(state, action) {
  const timeLeftSec = action.timeLeftMs / 1000;

  // FIXME: we should let each UI component decide what to do with this event
  // eslint-disable-next-line no-alert
  window.confirm(
    `Your session will time out in ${prettifyNumber(timeLeftSec)} seconds. Please save your work`);

  return state;
}
