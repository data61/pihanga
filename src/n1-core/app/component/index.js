
// Import all top level modules for web pack to find
import './index.css';

import { ACTION_TYPES as APP_ACTION_TYPES } from './app.actions';

import initReducers from './app.reducers';
import { APP_ROUTING } from './app.routing';

export { default as AppComponent } from './app.component';
export { reloadBackend, navigateToPage, createScratch } from './app.actions';

export const ACTION_TYPES = {
  ...APP_ACTION_TYPES,
};

/**
 * Register all reducers to interact between components
 */
export function init(register) {
  initReducers(register.reducer);
  register.routing(APP_ROUTING);
}
