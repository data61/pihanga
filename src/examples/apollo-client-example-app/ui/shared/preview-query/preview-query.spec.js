import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { graphql } from 'react-apollo';
import { previewQuery } from './preview-query.component';

jest.mock('react-apollo', () => ({
  graphql: jest.fn()
}));

describe('preview-query.component.jsx', () => {
  describe('#previewQuery', () => {
    describe('snapshots', () => {
      it('renders child HOC when not loading', () => {
        const mockHoc = () => <div />;
        graphql.mockReturnValue(child => child({ loading: false }));
        const wrapper = shallow(previewQuery('gqlString')(mockHoc));
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      it('renders spinner when loading', () => {
        const mockHoc = () => <div />;
        graphql.mockReturnValue(child => child({ loading: true }));
        const wrapper = shallow(previewQuery('gqlString')(mockHoc));
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });

    it('calls #graphql with the argument queryStr and options', () => {
      const mockOptions = jest.fn();
      previewQuery('gqlString', mockOptions)(() => <div />);
      expect(graphql).toHaveBeenCalledWith('gqlString', {
        options: mockOptions,
        props: expect.any(Function)
      });
    });
  });
});
