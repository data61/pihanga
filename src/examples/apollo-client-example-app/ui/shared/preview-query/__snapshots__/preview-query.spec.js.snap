// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview-query.component.jsx #previewQuery snapshots renders child HOC when not loading 1`] = `
<mockHoc
  loading={false}
/>
`;

exports[`preview-query.component.jsx #previewQuery snapshots renders spinner when loading 1`] = `
<Spinner
  loading={true}
/>
`;
